components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -0,0 +1,773 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.endpoint.ciba;
+
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.Payload;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth.ciba.common.CibaConstants;
+import org.wso2.carbon.identity.oauth.ciba.exceptions.ErrorCodes;
+import org.wso2.carbon.identity.oauth.ciba.model.CibaAuthCodeRequest;
+import org.wso2.carbon.identity.oauth.common.OAuth2ErrorCodes;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth.endpoint.exception.CibaAuthFailureException;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.identity.openidconnect.model.Constants;
+
+import java.text.ParseException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * Handles the validation of ciba authentication request.
+ */
+public class CibaAuthRequestValidator {
+
+    private static final Log log = LogFactory.getLog(CibaAuthRequestValidator.class);
+
+    private CibaAuthRequestValidator() {
+
+    }
+
+    private static CibaAuthRequestValidator cibaAuthRequestValidatorInstance = new CibaAuthRequestValidator();
+
+    public static CibaAuthRequestValidator getInstance() {
+
+        return cibaAuthRequestValidatorInstance;
+    }
+
+    /**
+     * Validate CIBA Authentication Request.
+     *
+     * @param request CIBA Authentication Request.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Exception.
+     * @throws CibaAuthFailureException CIBA server serror.
+     */
+    public void validateAuthRequestParams(String request) throws CibaAuthFailureException {
+
+        try {
+            long timeInMillis = Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC)).getTimeInMillis();
+            long skewTime = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() *
+                    CibaConstants.SEC_TO_MILLISEC_FACTOR;
+            SignedJWT signedJWT = SignedJWT.parse(request);
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+
+            if (!isValidSignature(signedJWT)) {
+                // Signature is invalid.
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT, "invalid signature.");
```

Comment:
Start the error message with a  capital letter.

Shouldn't the error code be an invalid_request?